app/controllers/application_controller.rb

```diff
@@ -367,6 +367,7 @@ def report_data
     if options.nil? || options[:view].nil?
       model_view = process_params_model_view(params, options)
       @edit = session[:edit]
+      options_from_session(options)
```

Comment:
> One thing that surprised me here as I was debugging get_view and report_data was the fact that we call get_view twice!! ... From nested_list and then from report_data. Why?

I have noticed that too. For the first page of data the call is realy done twice. Once when the grid component is rendered and 2nd time when the data is retrieved and passed as JSON. Subsequest (pagination) reguest call just the `report_data` endpoint so the query is fired once.  But have not dug deeper. @karelhala might know. 

Anyway the `report_data` endpoint is meant as a transition step toward the GTL component consuming some report API. Therefor I am concerned about adding more logic and exceptions there. I'd rather move in the opposite direction and cleanup and unify the different cases.